### DISCUSSION TITLE
Possible to provide a camera entity to Frigate instead of an RTSP URL?

### DISCUSSION THREAD

**Discussion by dspears0809** [blakeblackshear/frigate#2473] [Ask A Question]
Is it possible to provide a camera entity to Frigate instead of an RTSP URL?

I'd like to use the built-in "camera" entity of HA to proxy my camera streams and have frigate and lovelace live view refer to those proxied streams.  The idea being, I don't want to have Frigate or another addon handling the proxy responsibilities - just HA with one connection per stream to my cameras, and the rest referencing those camera entities.

**Comment by blakeblackshear** [comment]
It's not possible, but it may be in the future.

**Reply by dspears0809** [reply]
ok, thanks for the consideration :-)

**Reply by felipecrs** [reply]
This would be especially useful for Tuya-based cameras, which now shows on HA with the official Tuya integration, but do not provide a RTSP stream. Mainly all their doorbells as well.

**Reply by SomeoneSinister** [reply]
It would be really nice to have a one stop shop for all my camera events and recordings.
Currently some of my camera entities can only be used through a HA integration and doesnt support a RSTP for frigate

**Reply by blakeblackshear** [reply]
It's a limitation on the home assistant side. Home assistant doesn't provide a way to get the stream information in a way frigate can consume it.

**Comment by felipecrs** [comment]
It's now possible, thanks to a custom integration and go2rtc:

https://github.com/felipecrs/hass-expose-camera-stream-source#importing-home-assistant-cameras-to-go2rtc-andor-frigate
